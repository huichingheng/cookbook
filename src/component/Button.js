import React from "react";

const Button = props => {
  return (
    <button className={props.className} onClick={props.handleClick}>
      {props.content}
    </button>
  );
};

export default Button;
