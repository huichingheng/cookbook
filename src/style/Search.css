.search {
  background: tomato;
  padding: 6px 0;
  border-bottom: 1px solid black;
}
.search-input {
  /* background: blueviolet; */
  width: 60%;
}

.search-button {
    margin-left: 10px;
  }
.search-button:hover {
    background: whitesmoke;
    color: darkgray;
    box-shadow: blue;
    border-color: azure
  }
  
.classname:active {
  position: relative;
  top: 1px;
}
