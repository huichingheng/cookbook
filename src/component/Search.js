import React from "react";
import Button from "./Button";
import '../style/Search.css'
const INPUT_PLACEHOLDER = "Ingredients e.g. chicken, carrot, pasta...";
const SEARCH_BUTTON_TEXT = "Find Recipe";

const Search = props => {
  const { handleUserInput, handleSearchClick } = props;
  return (
    <div className="search">
      <input
        className="search-input"
        type="text"
        placeholder={INPUT_PLACEHOLDER}
        onChange={event => handleUserInput(event)}
      />
      <Button
        className="search-button"
        content={SEARCH_BUTTON_TEXT}
        handleClick={handleSearchClick}
      />
    </div>
  );
};

export default Search;
