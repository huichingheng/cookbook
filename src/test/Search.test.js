import React from "react";
import Search from "../component/Search";
import Button from "../component/Button";
import { shallow } from "enzyme";

const mockEventHandler = jest.fn(() => {});

const wrapper = shallow(<Search handleUserInput={mockEventHandler} />);
const inputBox = wrapper.find(".search-input");
const button = wrapper.find(Button);

describe("<Search /> should have", () => {
  it("An Input Field", () => {
    expect(inputBox).toHaveLength(1);
  });

  it("A Button Component", () => {
    expect(button).toHaveLength(1);
  });
});

describe("search-input", () => {
  it("should handle input events", () => {
    const TEST_VALUE = "test value";
    inputBox.simulate("change", { target: { value: TEST_VALUE } });
    expect(mockEventHandler).toBeCalledWith({
      target: { value: TEST_VALUE }
    });
  });
});
