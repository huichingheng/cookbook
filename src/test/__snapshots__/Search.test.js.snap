// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should match 1`] = `
Object {
  "children": Array [
    <input
      className="search-input"
      onChange={[Function]}
      placeholder="Ingredients e.g. chicken, carrot, pasta..."
      type="text"
/>,
    <Button
      className="search-button"
      content="Find Recipe"
      handleClick={undefined}
/>,
  ],
  "className": "search",
}
`;
